### PR TITLE
Bugfix renaming single file analysis

### PR DESCRIPTION
**Pull request by pmisteliac**

### CODE REVIEW

File: nabl2.runtime/trans/nabl2/runtime/transform/renaming.str [hunk 1]
```diff
@@ -208,18 +208,19 @@ rules // Utility strategies
     where
       new-name := <show-input-dialog(|"Rename", "")> "Enter a new name"
 
-  get-project-analysis: _ -> analysis-object
+  nabl2-get-project-analysis(multi): ast -> analysis
     where
       project-analyses := <get-project-constraint-analyses>
```
Comment (AZWN) [inline nabl2.runtime/trans/nabl2/runtime/transform/renaming.str:213]: I think you could move this operation in the `if` as well.

File: statix.runtime/trans/statix/runtime/renaming.str [hunk 2]
```diff
@@ -207,18 +207,19 @@ rules // Utility strategies
     where
       new-name := <show-input-dialog(|"Rename", "")> "Enter a new name"
 
-  get-project-analysis: _ -> analysis-object
+  get-project-analysis(multi): ast -> analysis
     where
       project-analyses := <get-project-constraint-analyses>
```
Comment (AZWN) [inline statix.runtime/trans/statix/runtime/renaming.str:212]: Same here

**Review by AZWN — CHANGES_REQUESTED** [review]
I'm not very familiar with this code, but the changes look reasonable. As a general remark, you might want to use `with` instead of `where` when a particular transformation must succeed. Even when under a choice operator, `with` will prevent backtracking, which often leads to better error messages/traces.

File: nabl2.runtime/trans/nabl2/runtime/transform/renaming.str (hunk 1 above)
Comment (pmisteliac) [reply nabl2.runtime/trans/nabl2/runtime/transform/renaming.str]: Done

File: statix.runtime/trans/statix/runtime/renaming.str (hunk 2 above)
Comment (pmisteliac) [reply statix.runtime/trans/statix/runtime/renaming.str]: Done

**Pull request merged**